src/libexpr/meson.build

```diff
@@ -0,0 +1,169 @@
+# Nix lib expr build file
+#============================================================================
+
+libexpr_dir = meson.current_source_dir()
+libexpr_inc = proj_inc
+
+
+# dependencies
+#============================================================================
+
+libexpr_dep_list = [
+    boost_dep,
+    libdl_dep,
+    libsodium_dep,
+    gc_dep,
+
+    libutil_dep,
+    libstore_dep,
+    libmain_dep,
+    libfetchers_dep]
+
+
+# src files
+#============================================================================
+
+libexpr_src = files(
+    join_paths(libexpr_dir, 'attr-path.cc'),
+    join_paths(libexpr_dir, 'attr-set.cc'),
+    join_paths(libexpr_dir, 'common-eval-args.cc'),
+    join_paths(libexpr_dir, 'eval-cache.cc'),
+    join_paths(libexpr_dir, 'eval.cc'),
+    join_paths(libexpr_dir, 'function-trace.cc'),
+    join_paths(libexpr_dir, 'get-drvs.cc'),
+    join_paths(libexpr_dir, 'json-to-value.cc'),
+    join_paths(libexpr_dir, 'nixexpr.cc'),
+    join_paths(libexpr_dir, 'primops.cc'),
+    join_paths(libexpr_dir, 'value-to-json.cc'),
+    join_paths(libexpr_dir, 'value-to-xml.cc'))
+
+
+libexpr_headers = files(
+    join_paths(libexpr_dir, 'attr-path.hh'),
+    join_paths(libexpr_dir, 'attr-set.hh'),
+    join_paths(libexpr_dir, 'common-eval-args.hh'),
+    join_paths(libexpr_dir, 'eval-cache.hh'),
+    join_paths(libexpr_dir, 'eval.hh'),
+    join_paths(libexpr_dir, 'eval-inline.hh'),
+    join_paths(libexpr_dir, 'function-trace.hh'),
+    join_paths(libexpr_dir, 'get-drvs.hh'),
+    join_paths(libexpr_dir, 'json-to-value.hh'),
+    join_paths(libexpr_dir, 'nixexpr.hh'),
+    join_paths(libexpr_dir, 'primops.hh'),
+    join_paths(libexpr_dir, 'symbol-table.hh'),
+    join_paths(libexpr_dir, 'value.hh'),
+    join_paths(libexpr_dir, 'value-to-json.hh'),
+    join_paths(libexpr_dir, 'value-to-xml.hh'))
+
+
+# include directories
+#========================================================================
+
+# include current dir
+#---------------------------------------------------
+libexpr_inc += include_directories('.')
+
+
+# add subdirectories
+#---------------------------------------------------
+libexpr_dirs = [
+  'flake',
+  'primops']
+
+foreach dir : libexpr_dirs
+    subdir(dir)
+endforeach
+
+
+# targets
+#============================================================================
+
+libexpr_src += custom_target(
+    'parser_tab.[cchh]',
+    output : [
+        'parser-tab.cc',
+        'parser-tab.hh'],
+    input : 'parser.y',
+    command : [
+        bison,
+        '-v',
+        '--output=@OUTPUT0@',
+        '@INPUT@',
+        '-d'])
```

Comment:
shouldn't that be:
```meson
'--defines', '@OUTPUT1@'])
```

Reply:
maybe, but it works currently.

CC @Ericson2314, i think this snippet is from your port attempt

Reply:
I certainly defer to @dcbaker with these things.

Reply:
It's just more explicit, apparently they do the same thing according to the man page